src/controllers/user.controller.ts

```diff
@@ -87,4 +160,56 @@ export class UserController {
   ): Promise<Product[]> {
     return this.recommender.getProductRecommendations(userId);
   }
+
+  @post('/users/login', {
+    responses: {
+      '200': {
+        description: 'Token',
+        content: {
+          'text/plain': {
+            schema: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    },
+  })
+  async login(
+    @requestBody() credentials: Credentials,
+  ): Promise<String | undefined> {
+    // Validate Email
+    if (!isemail.validate(credentials.email)) {
```

Comment:
Please move the implementation of `login` into `UserRepository` class, so that new tokens can be created from TS/JS API directly (e.g. in integration tests), without the need to go through the REST API.

Reply:
Uh oh, was the implementation of login moved back to the controller? I consider that as anti-pattern, it breaks separation of concerns in MVC model:
 - Models/Repositories are responsible for data manipulation and domain model
 - Controllers are responsible for REST-specific behavior
 - Views are rendering the HTML output. In our case, we are building REST/JSON APIs, the view function is `JSON.stringify`.

Regarding login:
 - The repository layer should be responsible for preforming the actual login. There may be other scenarios besides REST where we want to login: integration tests, gRPC transport, websocket transport, etc.
 - The (REST) controller is responsible for obtaining login parameters from the request, calling the repository method to perform the actual login, and then converting the result into an HTTP response.

cc @raymondfeng